docs/migration-v5-to-v6.md

```diff
@@ -0,0 +1,109 @@
+---
+id: migration-v5-to-v6
+title: Migration from v5 to v6
+---
+
+Starting with v6.0.0, the approach for doing validations is more declarative and in turn some of the ways we had done in v5 would require some changes. The purpose of this documetation is to show how to migrate our existing express-validator code from v5 to v6 through sample codes. With the understanding that there is a more declarative way of doing the validations in v6+, the only requirement in the sample codes we set for is to keep the similar programmatical approach in both versions so that we are comparing apples to apples.
+
+> **For a complete list of breaking changes and new features in 6.0.0**,
+> please check [its release notes](https://github.com/express-validator/express-validator/releases/tag/v6.0.0).
+
+## Sample code using v5.3.1
+Say we want to leverage this library in a more _programmatic_ way: A POST request handler takes a JSON object in the request's body and responds based on if the **_greetings_** attribute is found in the JSON object.
+```js
+const express = require('express');
+const bodyParser = require('body-parser');
+const expressValidator = require('express-validator');
+
+const app = express();
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(expressValidator());
+
+app.post('/', (req, res) => {
+  console.info(`checking request body: ${JSON.stringify(req.body)}`);
+  req.checkBody('greetings').isLength({min: 1});
+
+  const validationResults = req.validationErrors();
+  console.info(`validation results: ${JSON.stringify(validationResults)}`);
+
+  if (!validationResults || validationResults.length < 1) {
+    req.sanitizeBody('greetings').escape().trim();
+    res.send(`checking done, hello: ${req.body['greetings']}`);
+  } else {
+    res.send(`checking done, error: ${JSON.stringify(validationResults)}`);
+  }
+});
+
+if (module === require.main) {
+  const port = 3000;
+  app.listen(port, err => {
+    if (err) {
+      throw err;
+    }
+    console.info(`server listening at ${port}`);
+  });
+}
```

Comment:
> Why you chose the imperative way for v6?

For pragmatic reason and in migrating "legacy" projects mostly. There are use cases where we have to migrate from v5 code as quickly as possible and this can provide almost 1-to-1 parity. Totally agree with you that declarative approach is a cleaner way to go. And I see great documentations have been put together on this approach already like #871.  Afterall, I do it in all my newer projects that started with v6. I just see values in both approaches where we can decide and pick the best one for the job.